packages/core-microservice/tests/index.test.js

```diff
@@ -88,3 +122,266 @@ describe("#Microservice", () => {
     }
   });
 });
+
+describe("#Microservice - Server lifecycle", () => {
+  let CURRENT_PID;
+
+  beforeEach(() => {
+    CURRENT_PID = null;
+  });
+
+  afterEach(() => {
+    return new Promise((resolve) => {
+      kill(CURRENT_PID, "SIGTERM", resolve);
+    });
+  });
+
+  test("Run and shutdown microservice (not silent)", async () => {
+    const serverFilename = path.resolve(__dirname, "fixture", "server.js");
+    const addDebugLog = jest.fn();
+    const options = {
+      port: 9090,
+      command: "node " + serverFilename,
+      silent: false,
+      timeout: 1000,
+      state: {
+        outputStream: {
+          addDebugLog
+        }
+      }
+    };
+    const Instance = new Microservice(options);
+    expect(Instance.isRunning).toBe(false);
+    await Instance.start();
+    CURRENT_PID = Instance.server.pid;
+    expect(Instance.isRunning).toBe(true);
+    await Instance.stop();
+    expect(addDebugLog).toHaveBeenCalledWith(
+      "Server running on the port 9090\n"
+    );
+    expect(Instance.server.killed).toBe(true);
+  });
+
+  test("Run and shutdown microservice (silent)", async () => {
+    const serverFilename = path.resolve(__dirname, "fixture", "server.js");
+    const addDebugLog = jest.fn();
+    const options = {
+      port: 9090,
+      command: "node " + serverFilename,
+      silent: true,
+      timeout: 1000,
+      state: {
+        outputStream: {
+          addDebugLog
+        }
+      }
+    };
```

Comment:
I think we could abstract this into a helper function, as a lot of these properties don't change a lot.

```js
function buildMicroserviceOptions(fixtureName, overridedOptions) {
    const serverFilename = path.resolve(__dirname, "fixture", fixtureName");
    const addDebugLog = jest.fn();
    const options = {
      port: 9090,
      command: "node " + serverFilename,
      silent: true,
      timeout: 1000,
      state: {
        outputStream: {
          addDebugLog
        }
      },
      ...overridedOptions
    };

   return options
}
```

It's more an idea it could be done into another PR anyway